### PR TITLE
Compose for Web: Initial IME support

### PR DESCRIPTION
**Pull request by jershell**
## Proposed Changes
This PR is an attempt to provide basic support and accumulate knowledge about the state of affairs with keyboard (ime) in mobile web browsers

## Testing
chrome on mobile device - ok
safari on mobile device - ok

Test: 
Desktop browsers are not affected.
On mobile devices the keyboard is visible and works =\

## Google CLA
OK

## Working demo
https://github.com/JetBrains/compose-multiplatform-core/assets/1238301/edc518c3-6f18-428a-ab6d-4f19e283106a

## Demo problems
https://github.com/JetBrains/compose-multiplatform-core/assets/1238301/51751ee5-3ee4-4de7-9fbb-da6e3b944180

## Comments
    `<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=0">`
user-scalable=0 set to 0 to prevent default zoom in safari browser

## Limitations
  - Bad text selection
  - Context menu.
  > [!NOTE]  
  > Based on the information at this time, several implementation options are likely.
  > - By precise positioning, highlight content inside html input, provoking the browser to show the system context menu.
  > - Implement the context menu by compose.
  - Poor positioning to focused input.
> [!NOTE]  
> The keyboard changes the current viewPort, which does not work exactly, perhaps on google chrome you can use https://developer.chrome.com/blog/viewport-resize-behavior, but it is not clear what to do with safari
- safari doesn't show keyboard when clicking on focused TextField again.

## Additional. Why is the VirtualKeyboard API not used.
Summary of the VirtualKeyboard API https://w3c.github.io/virtual-keyboard/ at 2024
  - Specification being incubated in a Community Group
  - support is only available in [chrome](https://issues.chromium.org/issues?q=is:open%20VirtualKeyboard%20API),  
  - unsupported in [firefox](https://bugzilla.mozilla.org/show_bug.cgi?id=1730568)
  - unsupported in [safary](https://bugs.webkit.org/show_bug.cgi?id=230225)
  - requires an HTML element with the contenteditable and `virtualkeyboardpolicy="manual"` attributes.
  - There are problems and questions regarding use (https://github.com/w3c/virtual-keyboard/issues)
  
> [!NOTE]    
> [browser_compatibility](https://developer.mozilla.org/en-US/docs/Web/API/VirtualKeyboard_API#browser_compatibility,)

### CODE REVIEW

**Comment by jershell** [comment]
@eymar Hello. Can you help me with this PR?

File: compose/ui/ui/src/jsWasmMain/kotlin/androidx/compose/ui/window/ComposeWindow.js.kt [hunk 1]
```diff
@@ -159,6 +180,7 @@ private class ComposeWindow(
             event.preventDefault()
         })
 
+        // TODO Shouldn't we use e.preventDefault() to replace browser`s hotkeys(Tab, Ctrl+S, etc)?
```
Comment (Schahen) [inline compose/ui/ui/src/jsWasmMain/kotlin/androidx/compose/ui/window/ComposeWindow.js.kt]: Please, avoid adding comments to PR just to discuss any - however valid - topics, it's just the wrong tool for the purpose

**Comment by jershell** [comment]
@eymar Hello, is there any news on my PR?

**Comment by eymar** [comment]
Hi @jershell ! Thanks for a ping. The feature you implemented is really important and we appreciate your contribution :) 

We would like to carefuly evaluate this solution with its nuances and the possible alternatives if they are necessary.
Currently, the team is preparing (planning) for the next milestone. We'll make sure this feature is included. 
That means we'll review this PR properly a bit later.

**Comment by jershell** [comment]
@eymar Hello, any news?

**Comment by eymar** [comment]
> @eymar Hello, any news?

No news yet :) 
We will not forget about this PR - I can promise that. This feature is a must-have for 1.6.10 release.

**Comment by jershell** [comment]
I pulled latest master. Now we have a bug. The click does not hit on widget. I mention that the bug is not mine -_- and the keyboard works, you just need to press higher (about 50-60dp).

**Comment by jershell** [comment]
@eymar Hello. I see version 1.6.10*** got it. Does this mean that we can soon merge this PR?

**Comment by ismai117** [comment]
@jershell does the keyboard now show up if I use v1.6.10-dev1551?

**Comment by jershell** [comment]
> @jershell does the keyboard now show up if I use v1.6.10-dev1551?

no, only after the merger

**Comment by jershell** [comment]
@eymar hi, any news?

**Comment by eymar** [comment]
> @eymar hi, any news?

@Schahen is working on this topic (related parts).

**Comment by Schahen** [comment]
@jershell Hi jershell, I'd had to recompose you PR due to the changes in our event-related approach in general, but don't think that your contribution is wasted - it's actually quite the opposite, we've based our changes on the work you provided (and make sure that the authorship will be preserved in the final merge that's yet to be)

you can check https://github.com/JetBrains/compose-multiplatform-core/pull/1259 for further progress
this PR I am closing.

**Pull request closed without merging**